### PR TITLE
Initial README

### PR DESCRIPTION
**Pull request by GwendolenLynch**
Stolen from the PR, so I put your paws over the original lines @CarsonF .

~~Only grammar and positioning (and Markdown) changed, info is all the same, but a starting point!~~

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,304 @@
+Bolt Collection
+---------------
+
+
+## `Bag` and `ImmutableBag`
+
+These are Object Orientented implementations of arrays.
```
Comment (xiaohutai) [inline README.md]: `Object Oriented` -> `object-oriented`

File: README.md [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+    removeFirst(): mixed|null
+    removeLast(): mixed|null
+```
+</details>
+
+
+## What's `ImmutableBag` for?
+
+All of the methods in `ImmutableBag` are queries of the current state or return
+a _new_ modified Bag. Thus an `ImmutableBag` cannot be mutated...no duh. `Bag`
```
Comment (xiaohutai) [inline README.md]: I'd remove the `...no duh` part tbh 😋

**Review by xiaohutai — COMMENTED** [review]
some small nitpicky things 😋

File: README.md (hunk 1 above)
Comment (GwendolenLynch) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (GwendolenLynch) [reply README.md]: Done

Comment (xiaohutai) [reply README.md]: Dammmn.. You're way too fast :+1:

File: README.md [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+This method converts all traversable objects at any level to arrays in the
+return value.
+
+
+## Hasn't this been done already?
+
+Obviously others think PHP arrays suck as well and have attempted to resolve 
+this.
+ 
+Symfony's `ParamaterBag` is a good basic _map_ (associative arrays) container
```
Comment (bobdenotter) [inline README.md]: Typo: ParameterBag

**Review by CarsonF — APPROVED** [review]

**Pull request merged**